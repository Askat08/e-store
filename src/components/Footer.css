a {
  color: white;
  text-decoration: none;
}

a:hover {
  color: greenyellow;
  text-decoration: none;
}
