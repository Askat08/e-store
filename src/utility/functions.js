// Footer button scroll to top method
export const scrollToTop = () => {
  window.scrollTo({ top: 0, left: 0, behavior: "smooth" });
};

// handles local storage data
export const setLocalStorage = (array) => {
  // Array from local storage
  const localStorageArray =
    JSON.parse(window.localStorage.getItem("cart-items")) || [];

  // New array with no duplicates
  const newFilteredArray = array.reduce(
    (newArr, eachArrayElem) => {
      if (
        localStorageArray.findIndex(
          (eachStorageElem) => eachStorageElem.id === eachArrayElem.id
        ) === -1
      ) {
        newArr.push(eachArrayElem);
      }
      return newArr;
    },
    [...localStorageArray]
  );
  // set items to localstorage
  localStorage.setItem("cart-items", JSON.stringify(newFilteredArray));
};

// Get item from an array by it's id
export const getItembyId = (id, arr) => {
  const comic = arr.find((item) => item.id === id);
  return comic;
};

// modify count of item in an array
export const cartItemModifyCount = (arr, id, action, cart) => {
  let comics = [...arr];
  const index = comics.indexOf(getItembyId(id, comics));
  const item = comics[index];

  if (action === "dec" && item.count > 1) {
    item.count -= 1;
  } else if (action === "inc" && item.count < 10) {
    item.count += 1;
  } else if (action === "add") {
    if (item.count === undefined) {
      item.count = 1;
    } else if (item.count < 10 && item.count !== undefined) {
      item.count += 1;
    }
  }

  // Filtered array with incremented item added
  const arrayWithModifiedItem = [
    ...cart.concat(item).filter((v, i, a) => a.indexOf(v) === i),
  ];

  return arrayWithModifiedItem;
};
