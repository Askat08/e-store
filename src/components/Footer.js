import React from "react";
import "./Footer.css";

export default function Footer({ props, scrollToTop }) {
  return (
    <div className="sticky-bottom">
      <div className="card-footer bg-dark text-light">
        <div className="row">
          <h3 className="col-6 my-3">Marvel Comic Books</h3>
          <h4 className="card-text my-3 col-6 text-right">
            Get your superhero today.
          </h4>
        </div>
        <div className="row">
          <h5
            className="col-6 my-3 "
            dangerouslySetInnerHTML={{ __html: props }}
          ></h5>
          <h5 className="col-6 my-3 text-right">
            Coded with <i className="fas fa-laptop-code text-info"></i> by{" "}
            <a href="https://askatb.com" className="text-danger">
              me
            </a>
            .
          </h5>
        </div>
        <button
          className="btn btn-outline-info shadow-none"
          onClick={() => scrollToTop()}
        >
          top
        </button>
      </div>
    </div>
  );
}
