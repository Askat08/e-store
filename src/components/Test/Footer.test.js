import React from "react";
import { shallow } from "enzyme";
import Footer from "../Footer";
import * as methods from "../../utility/functions";

describe("Footer component test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });

  it("should match to snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render title of app", () => {
    const h3 = wrapper.find("h3");
    const title = h3.text();

    expect(title).toBe("Marvel Comic Books");
  });

  it("should render slogan of the app", () => {
    const h4 = wrapper.find("h4");
    const text = h4.text();

    expect(text).toBe("Get your superhero today.");
  });

  it("should include website link", () => {
    const anchorHref = wrapper.find("a").prop("href");

    expect(anchorHref).toEqual("https://askatb.com");
  });

  it("should test button click", () => {
    const mock = jest.spyOn(methods, "scrollToTop");
    mock.mockReturnValue("scrolled to top");

    wrapper.find("button").props().onClick();

    expect(methods.scrollToTop).toBeCalled();
    expect(methods.scrollToTop()).toBe("scrolled to top");

    mock.mockRestore();
  });
});
