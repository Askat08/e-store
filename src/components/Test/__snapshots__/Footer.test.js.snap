// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component test should match to snapshot 1`] = `
<div
  className="sticky-bottom"
>
  <div
    className="card-footer bg-dark text-light"
  >
    <div
      className="row"
    >
      <h3
        className="col-6 my-3"
      >
        Marvel Comic Books
      </h3>
      <h4
        className="card-text my-3 col-6 text-right"
      >
        Get your superhero today.
      </h4>
    </div>
    <div
      className="row"
    >
      <h5
        className="col-6 my-3 "
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
      />
      <h5
        className="col-6 my-3 text-right"
      >
        Coded with 
        <i
          className="fas fa-laptop-code text-info"
        />
         by
         
        <a
          className="text-danger"
          href="https://askatb.com"
          target="_blank"
        >
          me
        </a>
        .
      </h5>
    </div>
    <button
      className="btn btn-outline-info shadow-none"
      onClick={[Function]}
      type="button"
    >
      top
    </button>
  </div>
</div>
`;
